pyaedt/application/Analysis2D.py

```diff
@@ -74,7 +74,7 @@ def mesh(self):
     #     return self._post
 
     @aedt_exception_handler
-    def assignmaterial(self, obj, mat):
+    def assign_material(self, obj, mat):
```

Comment:
Whenever we change the API like this, we __must__ provide an old method that calls the new method and raise a warning (or deprecation error) for the user.  Also, it's best to provide a [Deprecated Directive](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated) in the docstring of the older method.

Pinging @jleonatti and @MaxJPRey as well.  Should we change the API, give the user a chance to change theirs too or at least be notified when it changes.  Otherwise, they'll simply stop upgrading, or worse, using our API due to stability concerns.  For this reason, I'm in favor of using the warning approach first and then using an error after a minor release or two.

Add the following above this:
```py
         def assignmaterial(self, obj, mat):
            """Assign a material to one or more objects.

            .. deprecated:: 0.3.1
               Use :func:`FieldAnalysis2D.assign_material` instead.

            """
            # one of the following:

            # raise a DeprecationWarning.  User won't have to change anything
            warnings.warn('assignmaterial is deprecated.  Please use assign_material instead',
                          warnings.DeprecationWarning)
            self.assign_material(obj, mat)

            # or raise an AttributeError (could also make a custom DeprecationError)
            raise AttributeError('assignmaterial is deprecated.  Please use assign_material instead')
```

Reply:
Added a "fun" write up regarding this here:
https://sphinxdocs.pyansys.com/coding_style.html

Reply:
I am being victim of my own suggestion 😄 
I'll do the change.